### PR TITLE
matching multiple choice between new and old apis

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: Describe your changes.
port multiple choice test to new FE api

**Motivation and Context**
new FE api

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_run_glue.py [hunk 1]
```diff
@@ -69,31 +69,45 @@ def setUp(self):
         self.logging_steps = 10
         self.rtol = 1e-02
 
+    def verify_old_and_new_api_are_equal(self, results_per_api):
+        new_api_results = results_per_api[True]
+        old_api_results = results_per_api[True]
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_glue.py:74]:
```suggestion
        old_api_results = results_per_api[False]
```
SHould this be `False`?

File: orttraining/orttraining/test/python/orttraining_run_multiple_choice.py [hunk 2]
```diff
@@ -91,22 +91,36 @@ def setUp(self):
...
+        expected_loss = 0.6665529619455844
 
-        results = self.run_multiple_choice(model_name="bert-base-cased", task_name="swag", fp16=True)
-        assert_allclose(results['acc'], expected_acc)
-        assert_allclose(results['loss'], expected_loss)
+        results_per_api = dict()
+        for use_new_api in [False, True]:
+            results = self.run_multiple_choice(model_name="bert-base-cased", task_name="swag", fp16=True, use_new_api=use_new_api)
+            assert_allclose(results['acc'], expected_acc)
+            assert_allclose(results['loss'], expected_loss)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_multiple_choice.py:117]:
```suggestion
```
Won't this be included into `results_per_api` which will then be tested by `verify_old_and_new_api_are_equal`

File: orttraining/orttraining/test/python/orttraining_run_multiple_choice.py [hunk 3]
```diff
@@ -91,22 +91,36 @@ def setUp(self):
...
+            results_per_api[use_new_api] = results
 
-    def run_multiple_choice(self, model_name, task_name, fp16):
+        for key in ['acc', 'loss']:
+            assert_allclose(results_per_api[False][key], results_per_api[True][key])
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_multiple_choice.py:121]:
```suggestion
        self.verify_old_and_new_api_are_equal(results_per_api)
```

File: orttraining/orttraining/test/python/orttraining_run_multiple_choice.py [hunk 4]
```diff
@@ -196,29 +211,45 @@ def compute_metrics(p: EvalPrediction) -> Dict:
 
         if model_name.startswith('bert'):
             model_desc = ModelDescription([
-                IODescription('input_ids', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=model.config.vocab_size),
-                IODescription('attention_mask', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=2),
-                IODescription('token_type_ids', [self.train_batch_size, num_labels, data_args.max_seq_length], torch.int64, num_classes=2),
-                IODescription('labels', [self.train_batch_size, num_labels], torch.int64, num_classes=num_labels)], [
+                IODescription('input_ids', ['batch', num_labels, 'max_seq_len_in_batch'], torch.int64, num_classes=model.config.vocab_size),
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_multiple_choice.py:214]: dtype and num_classes could be removed for the old API, right?

File: orttraining/orttraining/test/python/orttraining_run_multiple_choice.py (hunk 2 above)
Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_run_multiple_choice.py:117]: this test is to ensure run results match the golden number.

File: orttraining/orttraining/test/python/orttraining_run_glue.py [hunk 5]
```diff
@@ -128,7 +128,7 @@ def test_bert_with_mrpc(self):
                 results = self.run_glue(model_name="bert-base-cased", task_name="MRPC", fp16=False, use_new_api=use_new_api)
                 results_per_api[use_new_api] = results
 
-            self.verify_old_and_new_api_are_equal(results_per_api)
+            verify_old_and_new_api_are_equal(results_per_api)
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_glue.py:131]:
```suggestion
```
Line 158 already tests both if and else, right?

**Pull request merged**